content/ai-ml/ollama-openwebui/index.md

```diff
@@ -0,0 +1,285 @@
+   oc set volume deployment/open-webui --add --type=emptyDir --mount-path=/app/backend/static --name=static
+   oc create route edge --service=open-webui
+   ```
+
+## Verify deployment
+
+1. All nvidia pods should be running or completed
+   ```bash
+   oc get pods -n nvidia-gpu-operator
+   ```
+
+2. All pods of llm namespace should be running
+   ```bash
+   oc get pods -n llm
+   ```
+
+3. Check logs of ollama, it should detect inference compute card
+   ```bash
+   oc logs -l deployment=ollama
+   time=2024-09-12T07:28:40.446Z level=INFO source=images.go:753 msg="total blobs: 0"
+   time=2024-09-12T07:28:40.446Z level=INFO source=images.go:760 msg="total unused blobs removed: 0"
+   time=2024-09-12T07:28:40.446Z level=INFO source=routes.go:1172 msg="Listening on [::]:11434 (version 0.3.10)"
+   time=2024-09-12T07:28:40.446Z level=INFO source=payload.go:30 msg="extracting embedded files" dir=/tmp/ollama1403693285/runners
+   time=2024-09-12T07:28:53.779Z level=INFO source=payload.go:44 msg="Dynamic LLM libraries [cuda_v12 rocm_v60102 cpu cpu_avx cpu_avx2 cuda_v11]"
+   time=2024-09-12T07:28:53.779Z level=INFO source=gpu.go:200 msg="looking for compatible GPUs"
+   time=2024-09-12T07:28:54.324Z level=INFO source=types.go:107 msg="inference compute" id=GPU-51dedb8e-2306-b077-67c1-774b4206c8da library=cuda variant=v12 compute=7.5 driver=12.4 name="Tesla T4" total="14.6 GiB" available="14.5 GiB"
+   ```
+## Download a model
+
+1. Download llama3.1 8B using Ollama CLI
+   ```bash
+   oc exec svc/ollama -- ollama pull llama3.1
+   ```
+   You can check all models available on [https://ollama.com/library](https://ollama.com/library)
+
+## Accessing OpenWebUI
+
+After deploying OpenWebUI, follow these steps to access and configure it:
+
+1. Get the route URL:
+   ```bash
+   oc get route open-webui
+   ```
+
+2. Open the URL in your web browser. You should see the OpenWebUI login page. [https://docs.openwebui.com/](https://docs.openwebui.com/)
+
+3. Initial Setup:
+- The first time you access OpenWebUI, you'll need to register.
+- Choose a strong password for the admin account.
+
+4. Configuring Models:
+- Once logged in, go to the "Models" section to choose the LLMs you want to use.
+
+5. Testing Your Setup:
+- Create a new chat and select one of the models you've configured.
+- Try sending a test prompt to ensure everything is working correctly.
+
+6. Discover OpenWeb UI! You get lot of features like :
+- Model Builder
+- Local and Remote RAG Integration
+- Web Browsing Capabilities
+- Role-Based Access Control (RBAC)
+
+More here : [https://docs.openwebui.com/features](https://docs.openwebui.com/features)
+
+
+## Scale
+
+If you want to give best experience for multiple users, for improving response time and token/s you can scale Ollama app.
+
+Note that here you should use EFS (RWX access) instead or EBS (RWO access) for storage of ollama models, you can install EFS operator using [this tutorial](https://cloud.redhat.com/experts/rosa/aws-efs/)
```

Comment:
```suggestion
Note that here you should use the EFS (RWX access) storage class instead of the EBS (RWO access) storage class for the storage of ollama models. For instructions on how to set this up, please see [this tutorial](https://cloud.redhat.com/experts/rosa/aws-efs/)
```